### PR TITLE
postgres destination docker build & integration tests

### PR DESCRIPTION
**Pull request by jrhizor**
This sets ups a docker build in Gradle for the postgres destination and tests that image when on the master branch or when run locally.

There's a bit of code repetition with the BigQuery version. 

The input config doesn't match the image's expected config. I'm going to make the image accept the json config version and convert it to the target's config version in a future PR (which will require updating this test to use the correct input config). 

To build the postgres destination docker image locally:
`./gradlew :dataline-integration:singer:postgres:destination:docker:buildImage` 

To run the integration tests locally:
`./gradlew :dataline-integration:singer:postgres:destination:integrationTest`

### CODE REVIEW

**Review by sherifnada — CHANGES_REQUESTED** [review]
Putting this here for discussion since it may be beyond the scope of this PR, but the main concern I have with this PR is that it does not give us confidence that a connector works with its spec. This is a dangerous pattern because even if integration tests are passing, this doesn't assure us that a connector will work when called from the API. The current state of the PR is a good example where the name of the config fields accepted by the tap are different from the API's spec. Maybe we should discuss offline

**Comment by sherifnada** [comment]
From offline discussion: the desired end-state is to have an integration declare its own typed spec that is consumed by the API. This way, the API will not compile unless it passes the correct config fields expected by the integration. Will create an issue to track this separately. For now we'll use the exact naming declared by Singer.

**Comment by jrhizor** [comment]
After discussing a bit offline with Sherif, I'm changing the spec here to match what the singer tap expects. However, there still isn't a good way of having the test here actually fail until the integration provides the spec (which is what I'm working on now).

**Review by sherifnada — APPROVED** [review]

**Comment by jrhizor** [comment]
Merging for now. I'll update the tests with additional config validation later.

File: dataline-integrations/singer/postgres/destination/docker/build.gradle [hunk 1]
```diff
@@ -0,0 +1,10 @@
+plugins {
```
Comment (michel-tricot) [inline dataline-integrations/singer/postgres/destination/docker/build.gradle:1]: I don't think it is necessary to go all the way to the docker push here. This is more process piece here, as long as the docker file woks and the tests work (and soon have the config be part of it). 

There is already a push script for the integrations `tools/integrations/manage.sh` which i would love to see become the cli to create the integrations (with scaffolding etc...)

Comment (jrhizor) [reply dataline-integrations/singer/postgres/destination/docker/build.gradle:1]: Talked offline. We're going to keep this here to build for integration tests and we can use `manage.sh` for actually building and pushing images to an external registry.

**Pull request merged**